src/WingetCreateCore/Common/PackageParser.cs

```diff
@@ -334,6 +307,76 @@ public static Installer CloneInstaller(Installer installer)
             return JsonConvert.DeserializeObject<Installer>(json);
         }
 
+        /// <summary>
+        /// Finds an existing installer that matches the new installer by checking the installerType and the following:
+        /// 1. Matching based on architecture specified as an override.
+        /// 2. Matching based on architecture detected from URL string or binary.
+        /// </summary>
+        /// <param name="newInstaller">New installer to be matched.</param>
+        /// <param name="existingInstallers">List of existing installers to be matched.</param>
+        /// <param name="installerManifest">Installer manifest.</param>
+        /// <param name="detectedArchOfInstallers">List of detected architectures for each new installer.</param>
+        /// <param name="unmatchedInstallers">List of unmatched installers.</param>
+        /// <param name="multipleMatchedInstallers">List of installers with multiple matches..</param>
+        /// <returns>The installer match from the list of existing installers.</returns>
+        private static Installer FindInstallerMatch(
+            Installer newInstaller,
+            List<Installer> existingInstallers,
+            InstallerManifest installerManifest,
+            List<DetectedArch> detectedArchOfInstallers,
+            ref List<Installer> unmatchedInstallers,
+            ref List<Installer> multipleMatchedInstallers)
+        {
+            DetectedArch detectedArch = detectedArchOfInstallers.Find(
+                i => i.Url == newInstaller.InstallerUrl &&
+                (i.UrlArch == newInstaller.Architecture || i.BinaryArch == newInstaller.Architecture || i.OverrideArch == newInstaller.Architecture));
+
+            // If we can find an exact match by comparing the installerType and the override architecture, then return match.
+            // Otherwise, continue and try matching based on arch detected from url and binary detection, as the user could be trying overwrite with a new architecture.
+            var overrideArchMatches = existingInstallers.Where(
+                    i => (i.InstallerType ?? installerManifest.InstallerType) == newInstaller.InstallerType &&
+                    i.Architecture == detectedArch.OverrideArch);
+
+            if (overrideArchMatches.Count() == 1)
+            {
+                return overrideArchMatches.Single();
+            }
+
+            var urlArchMatches = existingInstallers.Where(
+                                        i => (i.InstallerType ?? installerManifest.InstallerType) == newInstaller.InstallerType &&
```

Comment:
[nit]: the tabbing here is weird and different than line 337

Reply:
fixed weird tab spacing